pro_tes/middleware/task_distribution/distance.py

```diff
@@ -46,8 +45,10 @@ def task_distribution(input_uri: List) -> List:
     # get the combination of the tes ip and input ip
     try:
         ips = ip_combination(input_uri=input_uri, tes_uri=tes_uri)
-    except (InputUriError, TesUriError):
-        return random.task_distribution()
+    except InputUriError as exc:
+        raise InputUriError from exc
+    except TesUriError as exc:
+        raise TesUriError from exc
```

Comment:
There's no point in doing that. Just let the error raise itself, implicitly! That is, remove the entire `try`-`except` here. Then you also do not need to document the exceptions in `Raises` in the docstring (which otherwise you would have needed to add, try to remember that).

Reply:
Understood, will fix it in the next commit.

Reply:
Done